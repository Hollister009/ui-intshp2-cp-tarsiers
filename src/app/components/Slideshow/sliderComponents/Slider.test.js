import React from 'react';
import { shallow, mount } from 'enzyme';
import Slider from './Slider';
import Images from './Slides';

describe('<Slider />', () => {
  let props;

  beforeEach(() => {
    props = { slides: [], activeIndex: 0, isStopped: false };
  });

  it('should match its snapshot', () => {
    const wrapper = shallow(<Slider {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle click slider indicator', () => {
    const wrapper = mount(<Slider {...props} />);

    wrapper.setProps({ onClick: 'handleClickIndicator' });
    expect(wrapper.props().onClick).toBe('handleClickIndicator');
  });

  it('should contain Image component', () => {
    const wrapper = shallow(<Slider {...props} />);

    expect(wrapper.find(<Images />).exists).toBeTruthy();
  });
});
