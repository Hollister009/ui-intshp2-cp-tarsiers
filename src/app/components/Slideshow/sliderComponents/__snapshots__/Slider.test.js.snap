// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should match its snapshot 1`] = `
<div
  className="slider-container"
>
  <div
    className="slider"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <SliderLeftArrow
      onClick={[Function]}
    />
    <ul
      className="slider__slides"
    />
    <SliderRightArrow
      onClick={[Function]}
    />
    <ul
      className="slider__indicators"
    />
  </div>
</div>
`;
